### PR TITLE
Add StateT transformS

### PR DESCRIPTION
**Pull request by adelbertc**

### CODE REVIEW

**Comment by milessabin** [comment]
:+1:

File: core/src/main/scala/cats/state/StateT.scala [hunk 1]
```diff
@@ -73,6 +73,20 @@ final class StateT[F[_], S, A](val runF: F[S => F[(S, A)]]) extends Serializable
   def transformF[G[_], B](f: F[(S, A)] => G[(S, B)])(implicit F: FlatMap[F], G: Applicative[G]): StateT[G, S, B] =
     StateT(s => f(run(s)))
 
+  /** Transform the state used
```
Comment (ceedubs) [inline core/src/main/scala/cats/state/StateT.scala]: Unjustifiably nitpicky: could you bump "Transform..." down to the next line to be consistent with the other ScalaDoc comments?

Also, what do you think about adding an sbt-doctest example here?

File: core/src/main/scala/cats/state/StateT.scala [hunk 2]
```diff
@@ -74,6 +74,33 @@ final class StateT[F[_], S, A](val runF: F[S => F[(S, A)]]) extends Serializable
...
+   * scala> type GlobalEnv = (Int, String)
+   * scala> val x: StateT[Option, Int, Double] = StateT((x: Int) => Option((x + 1, x.toDouble)))
+   * scala> val xt: StateT[Option, GlobalEnv, Double] = x.transformS[GlobalEnv](_._1, (t, i) => (i, t._2))
+   * scala> val input = 5
+   * scala> x.run(input)
+   * res0: Option[(Int, Double)] = Some((6,5.0))
+   * scala> xt.run((input, "hello"))
+   * res1: Option[(GlobalEnv, Double)] = Some(((6,hello),5.0))
+   * }}}
+   */
```
Comment (ceedubs) [inline core/src/main/scala/cats/state/StateT.scala]: Love it.

**Comment by ceedubs** [comment]
:+1:

**Pull request merged**